### PR TITLE
Use appropriate ellama prefix based on mode.

### PR DESCRIPTION
**Pull request by stephenwithav**
Depends on `ellama-major-mode` and `ellama-nick-prefix-depth`.

Currently supports org-mode, but defaults to markdown-mode's prefix character #.

Eliminates the need for ellama-nick-prefix and ellama-session-file-extension.

Solves #57.

### CODE REVIEW

**Comment by stephenwithav** [comment]
fyi, I keep force pushing to try to get it to recognize my pgp key.

**Comment by stephenwithav** [comment]
Oh well.

**Comment by s-kostyaev** [comment]
Hi @stephenwithav. Thank you for this PR.

File: ellama.el [hunk 1]
```diff
@@ -492,6 +487,22 @@ CONTEXT contains context for next request."
 
 (defvar ellama--new-session-context nil)
 
+(defun ellama-get-nick-prefix-for-mode ()
+  "Returns preferred header prefix char based om the current mode.
+Defaults to #, but supports org-mode. Depends on ellama-major-mode."
+  (let* ((preferred-mode (symbol-name ellama-major-mode))
+         (prefix-char
+          (cond ((string= preferred-mode "org-mode") ?*)
```
Comment (s-kostyaev) [inline ellama.el:495]: it's better to check `(derived-mode-p 'org-mode)` like here https://github.com/s-kostyaev/ellama/blob/90ec5e86bae8c0071f1f44c1ddbb14b62004b71d/ellama.el#L765

Comment (s-kostyaev) [reply ellama.el:495]: But this check should be evaluated in session buffer (I think it's already done).

File: ellama.el [hunk 2]
```diff
@@ -492,6 +487,22 @@ CONTEXT contains context for next request."
...
+                (t ?#))))
+    (make-string ellama-nick-prefix-depth prefix-char)))
+
+(defun ellama-get-session-file-extension ()
+  "Returns file extension based om the current mode.
+Defaults to md, but supports org. Depends on ellama-major-mode."
+  (let ((preferred-mode (symbol-name ellama-major-mode)))
+    (cond ((= preferred-mode "org-mode") "org")
```
Comment (s-kostyaev) [inline ellama.el:503]: Same as above.

**Comment by s-kostyaev** [comment]
@stephenwithav Please fix issues addressed in code review.

**Comment by stephenwithav** [comment]
Sorry for the delay...

File: ellama.el [hunk 3]
```diff
@@ -515,6 +510,20 @@ CONTEXT contains context for next request."
 
 (defvar ellama--new-session-context nil)
 
+(defun ellama-get-nick-prefix-for-mode ()
+  "Return preferred header prefix char based om the current mode.
+Defaults to #, but supports \"org-mode\".  Depends on \"ellama-major-mode.\""
+  (let* ((prefix-char
+          (cond ((provided-mode-derived-p ellama-major-mode "org-mode") ?*)
```
Comment (s-kostyaev) [inline ellama.el:517]:
```suggestion
          (cond ((provided-mode-derived-p ellama-major-mode 'org-mode) ?*)
```

File: ellama.el [hunk 4]
```diff
@@ -515,6 +510,20 @@ CONTEXT contains context for next request."
...
+                (t ?#))))
+    (make-string ellama-nick-prefix-depth prefix-char)))
+
+(defun ellama-get-session-file-extension ()
+  "Return file extension based om the current mode.
+Defaults to md, but supports org.  Depends on \"ellama-major-mode.\""
+  (cond ((provided-mode-derived-p ellama-major-mode "org-mode") "org")
```
Comment (s-kostyaev) [inline ellama.el:524]:
```suggestion
  (cond ((provided-mode-derived-p ellama-major-mode 'org-mode) "org")
```

File: ellama.el [hunk 5]
```diff
@@ -515,6 +510,20 @@ CONTEXT contains context for next request."
 
 (defvar ellama--new-session-context nil)
 
+(defun ellama-get-nick-prefix-for-mode ()
+  "Return preferred header prefix char based om the current mode.
+Defaults to #, but supports \"org-mode\".  Depends on \"ellama-major-mode.\""
```
Comment (s-kostyaev) [inline ellama.el:515]:
```suggestion
Defaults to #, but supports `org-mode'.  Depends on `ellama-major-mode'."
```

**Comment by s-kostyaev** [comment]
@stephenwithav Thank you!

**Pull request merged**